go/shared.go

```diff
@@ -1,3 +1,40 @@
 package app
 
+import (
+	"go.temporal.io/sdk/workflow"
+)
+
 const BreakfastTaskQueue = "BREAKFAST_TASK_QUEUE"
+
+type Compensations []any
+
+func (s *Compensations) AddCompensation(activity any) {
+	*s = append(*s, activity)
+}
+
+func (s Compensations) Compensate(ctx workflow.Context, inParallel bool) {
+	if !inParallel {
+		for i := len(s) - 1; i >= 0; i-- {
+			errCompensation := workflow.ExecuteActivity(ctx, s[i]).Get(ctx, nil)
+			if errCompensation != nil {
+				workflow.GetLogger(ctx).Error("Executing compensation failed", errCompensation)
```

Comment:
```suggestion
				workflow.GetLogger(ctx).Error("Executing compensation failed", "Error", errCompensation)
```

The values to the loggers have to be in key-value form (here and in other logger uses). Also, you can just use `err` variable. In Go, users are encouraged to use small variable names.